webapp/packages/plugin-data-viewer/src/ValuePanelPresentation/TextValue/useTextValue.ts

```diff
@@ -21,31 +14,59 @@ interface IUseTextValueArgs {
   currentContentType: string;
 }
 
-export function useTextValue({ model, resultIndex, currentContentType }: IUseTextValueArgs) {
-  const format = model.source.getAction(resultIndex, ResultSetFormatAction);
-  const editor = model.source.getAction(resultIndex, ResultSetEditAction);
+interface IUseTextValue {
+  textValue: string;
+  isFullTextValue: boolean;
+  isTruncated: boolean;
+  isTextColumn: boolean;
+}
```

Comment:
seems like you need to move load function to this hook (so, you can trigger full data loading from one place and use cache)

right now cache access and and loading is separated so we can't see the big picture